scripts/validate_docstrings.py

```diff
@@ -624,6 +625,12 @@ def get_validation_data(doc):
         errs.append(error('GL07',
                           correct_sections=', '.join(correct_order)))
 
+    pattern = re.compile('.. deprecated:: ')
+    if (bool(pattern.search(doc.summary))
+            or bool(pattern.search(doc.extended_summary))):
```

Comment:
do you mind creating a property `deprecated_with_directive` to avoid repeating this code?

also, not sure why we implemented it this way (and it was possibly me), but I think `'.. deprecated:: ' in (doc.summary + doc.extended_summary)` is a much simpler way to do it

Reply:
Sure I can do that,

I had a few doubts - did you mean to also change the current `deprecated` property to use the simpler `'.. deprecated:: ' in (doc.summary + doc.extended_summary)` implementation?

Also is there any special significance of the name `deprecated_with_directive`? Thanks!

Reply:
the current `deprecated` property looks for the methods/functions deprecated with the `.. deprecated::` directive, and also for anything that belongs to `Panel`.

So, the idea would be to have a property to check the cases deprecated with the directive (so, `deprecated_with_directive` could be a sensible name for it), and then the current `deprecated` property would use it, and also check for the `Panel` stuff.

Reply:
Got it, thanks!

Have updated both properties in this manner,